Tests/StreamChatTests/Controllers/PollsControllers/PollController_Tests.swift

```diff
@@ -0,0 +1,369 @@
+        XCTAssertNil(controller.poll)
+
+        // Create current user in the database
+        try client.databaseContainer.createCurrentUser(id: currentUserId)
+
+        // Create poll in that matches controller's `pollId`
+        let user = UserPayload.dummy(userId: currentUserId)
+        try client.databaseContainer.createPoll(id: pollId, createdBy: user)
+
+        // Assert message is fetched from the database and has correct field values
+        var poll = try XCTUnwrap(controller.poll)
+        XCTAssertEqual(poll.id, pollId)
+
+        // Simulate response from the backend with updated `name`, update the local poll in the database
+        let updatedName = "Poll Updated"
+        let pollPayload: PollPayload = XCTestCase().dummyPollPayload(
+            id: pollId,
+            name: updatedName,
+            user: user
+        )
+        try client.databaseContainer.writeSynchronously { session in
+            try session.savePoll(payload: pollPayload, cache: nil)
+        }
+
+        // Assert the controller's `poll` is up-to-date
+        poll = try XCTUnwrap(controller.poll)
+        XCTAssertEqual(poll.id, pollId)
+        XCTAssertEqual(poll.name, updatedName)
+    }
+    
+    func test_ownVotesAreFetched_afterCallingSynchronize() throws {
+        // Simulate `synchronize` call
+        controller.synchronize()
+        
+        // Create current user in the database
+        try client.databaseContainer.createCurrentUser(id: currentUserId)
+        
+        let user = UserPayload.dummy(userId: currentUserId)
+        try client.databaseContainer.createPoll(id: pollId, createdBy: user)
+
+        // Create own votes.
+        var ownVotes = [PollVotePayload]()
+        for _ in 0..<5 {
+            ownVotes.append(XCTestCase().dummyPollVotePayload(pollId: pollId, userId: user.id, user: user))
+        }
+        
+        let response = PollVoteListResponse(duration: "", votes: ownVotes)
+        let query = controller.ownVotesQuery
+        try client.databaseContainer.writeSynchronously { session in
+            try session.savePollVotes(payload: response, query: query, cache: nil)
+        }
+
+        XCTAssertEqual(controller.poll?.id, pollId)
+        XCTAssertEqual(controller.ownVotes.count, ownVotes.count)
+        
+        controller = nil
+        client = nil
+    }
+    
+    // MARK: - Casting votes
+    
+    func test_addVote_propagatesError() {
+        // Simulate `castPollVote` call and catch the completion.
+        var completionError: Error?
+        controller.castPollVote(answerText: nil, optionId: "123") {
+            completionError = $0
+        }
+
+        // Simulate network response with the error.
+        let networkError = TestError()
+        client.mockPollsRepository.castPollVote_completion?(networkError)
+
+        // Assert error is propagated.
+        AssertAsync.willBeEqual(completionError as? TestError, networkError)
```

Comment:
nit: Whenever we can, it would be nice not to use AssertAsync, and just use regular expectations. We agreed on moving out from custom frameworks in tests.